test/common/memory/memory_release_test.cc

```diff
@@ -59,17 +59,16 @@ class MemoryReleaseTest : public testing::Test {
 };
 
 TEST_F(MemoryReleaseTest, ReleaseRateAboveZeroDefaultIntervalMemoryReleased) {
-// TODO(nezdolik) Investigate and fix issue with gperf tcmalloc for memory release.
-// https://github.com/envoyproxy/envoy/issues/32447
-#if defined(GPERFTOOLS_TCMALLOC)
-  GTEST_SKIP() << "Skipping test, memory releasing is not yet supported for gperf tcmalloc.";
-#endif
   size_t initial_allocated_bytes = Stats::totalCurrentlyAllocated();
   auto a = std::make_unique<unsigned char[]>(MB);
   auto b = std::make_unique<unsigned char[]>(MB);
   if (Stats::totalCurrentlyAllocated() <= initial_allocated_bytes) {
     GTEST_SKIP() << "Skipping test, cannot measure memory usage precisely on this platform.";
   }
+#if defined(GPERFTOOLS_TCMALLOC)
+  EXPECT_DEATH(initialiseAllocatorManager(MB /*bytes per second*/, 0),
+               "Memory releasing is not supported for gperf tcmalloc.");
+#else
```

Comment:
Can we just elif the remaining sectionon TCMalloc?. e.g. 

```
#elif defined(TCMALLOC) 
  <code>
#endif     
````

If so we can get rid of the nested check making this a bit easier to read